src/store/modules/DownloadModule.ts

```diff
@@ -0,0 +1,44 @@
+import Vue from "vue";
+
+export interface DownloadProgress {
+    assignId: number;
+    initialMods: string[];
+    modName: string;
+    progress: number;
+    failed: boolean;
+}
+
+
+interface State {
+    allVersions: [number, DownloadProgress][],
+    assignId: number,
+    downloadObject: DownloadProgress | null,
+}
+
+/**
+ * State for mod downloads.
+ */
+export const DownloadModule = {
+    namespaced: true,
+
+    state: (): State => ({
+        allVersions: [],
+        assignId: 0,
+        downloadObject: null, // TODO: Check if the last element of allVersions can be used instead of this
+    }),
+
+    mutations: {
+        setDownloadObject(state: State, downloadObject: DownloadProgress) {
+            state.downloadObject = Object.assign({}, downloadObject);
+        },
+        pushDownloadObjectToAllVersions(state: State, params: { assignId: number, downloadObject: DownloadProgress }) {
+            state.allVersions.push([params.assignId, params.downloadObject]);
+        },
+        updateDownloadObject(state: State, params: { assignId: number, downloadObject: DownloadProgress }) {
+            Vue.set(state.allVersions, params.assignId, params.downloadObject);
```

Comment:
I think using the Vue method here is unnecessary, and not ideal from readability perspective (the reader might not know what exactly the set method does, but will probably know what the basic JS array methods do). And like above, just to be sure I'd rather create a new array rather than mutate the existing. Something like this (but do note the earlier comment about the arguments passed to this function not really matching the current argumen typing):

```
const newVersions = [...state.allVersions];
newVersions[params.assignId] = params.downloadObject;
state.allVersions = newVersions;
```